### PR TITLE
Fix/hck 4339 fe-ing to file it should also provide the ability to set a number of rows to generate

### PR DESCRIPTION
**Pull request by dannleed**
## Technical details

Added a new flag `generateSamplesInBulk` that controls the display of the input field for the number of generated samples in the FE modal window for the target script, if the "Include samples" checkbox is checked.
From the application in this case, a property `entitiesData` in the `data` argument is passed to the API methods for generating FE-ing scripts, containing the path to a file in the temporary folder with the generated samples. It is processed by the `getDataForSampleGeneration` function and, unless undefined, the samples will be generated by the `generateSamplesForEntity` function.

### CODE REVIEW

File: forward_engineering/types/coreApplicationTypes.js [hunk 1]
```diff
@@ -67,6 +67,11 @@ class CoreData {
      */
     entities
 
+    /**
+     * @type {any}
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/types/coreApplicationTypes.js:71]: Set type to EntitiesData

File: forward_engineering/helpers/feScriptBuilder.js [hunk 2]
```diff
@@ -34,7 +34,8 @@ const {getTableStatement} = require("./tableHelper");
 const {getIndexes} = require("./indexHelper");
 const {buildScript, getName, getTab, isSupportUnityCatalog, isSupportNotNullConstraints} = require("../utils/general");
 const {getViewScript} = require("./viewHelper");
-const {generateSampleForSeparateBucketTable, parseJsonData} = require('../sampleGeneration/sampleGenerationService');
+const {generateSamplesScript, getDataForSampleGeneration} = require('../sampleGeneration/sampleGenerationService');
+const { batchProcessFile } = require('../../reverse_engineering/helpers/fileHelper');
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/helpers/feScriptBuilder.js:38]: Reformat

File: forward_engineering/api.js [hunk 3]
```diff
@@ -28,10 +28,13 @@ const {
     Logger,
     PluginError
 } = require('./types/coreApplicationTypes')
-const {getSampleGenerationOptions, parseJsonData, generateSampleForDemonstration} = require("./sampleGeneration/sampleGenerationService");
+const {getSampleGenerationOptions, parseJsonData, generateSampleForDemonstration, generateSamplesScript} = require("./sampleGeneration/sampleGenerationService");
+const { batchProcessFile } = require('../reverse_engineering/helpers/fileHelper');
+const { getDataForSampleGeneration } = require('./sampleGeneration/sampleGenerationService');
```
Comment (ArzamastsevVladyslav) [inline forward_engineering/api.js:33]: Reformat

**Review by ArzamastsevVladyslav — DISMISSED** [review]

File: forward_engineering/api.js [hunk 4]
```diff
@@ -192,8 +206,32 @@ const getContainerScriptWithNotSeparateBuckets = (app, data,) => {
...
+        const demoSample = generateSamplesScript(_)(jsonSchema, [jsonData])
+        samples.push(demoSample)
+
+        await batchProcessFile({
+            filePath,
+            batchSize: 1,
+            parseLine: line => JSON.parse(line),
+            batchHandler: async (batch) => {
+                const sample = generateSamplesScript(_)(jsonSchema, batch);
+                samples.push(sample)
```
Comment (dannleed) [inline forward_engineering/api.js:229]: add logger

File: forward_engineering/api.js [hunk 5]
```diff
@@ -192,8 +206,32 @@ const getContainerScriptWithNotSeparateBuckets = (app, data,) => {
...
-    return getScriptAndSampleResponse(scripts, demoSample);
+    if (parsedData.isInvokedFromApplyToInstance) {
+        const demoSample = generateSampleForDemonstration(app, parsedData, 'container');
+
+        return getScriptAndSampleResponse(scripts, demoSample);
+    }
+
+    const samples = []
+
+    for (const entityData of Object.values(parsedData.entitiesData || {})) {
```
Comment (dannleed) [inline forward_engineering/api.js:217]: check for race condition

File: forward_engineering/config.json [hunk 6]
```diff
@@ -22,6 +22,9 @@
 	"applyToInstanceSettings": {
 		"generateSamplesInBulk": true
 	},
+	"forwardEngineeringSettings": {
+		"generateSamplesInBulk": true
```
Comment (dannleed) [inline forward_engineering/config.json:26]: move to the root level and update docs on gitbook

File: forward_engineering/helpers/feScriptBuilder.js [hunk 7]
```diff
@@ -135,30 +134,79 @@ const getContainerLevelViewScriptDtos = (data, _) => {
...
+            await batchProcessFile({
+                filePath,
+                batchSize: 1,
+                parseLine: line => JSON.parse(line),
+                batchHandler: async batch => {
+                    const sample = generateSamplesScript(_)(jsonSchema, batch);
+                    sampleScripts.push(sample);
+                },
+            });
+        }
```
Comment (dannleed) [inline forward_engineering/helpers/feScriptBuilder.js:177]: DRY

**Review by VitaliiBedletskyi — APPROVED** [review]

**Pull request merged**